### ISSUE TITLE
EAS / ActiveSync Issue with TLS1.2

### ISSUE THREAD

**Issue by epon-was-taken** [mailcow/mailcow-dockerized#2790]
Hello Guys,

I'm having an issue with the client connection on my Android phone.

When I am trying to add the mail account via Exchange I get two problems:
1. Autodiscover doesn't resolve the right server entry. It just resolves domain.com instead of mail.domain.com
2. When I correct the server entry to what it should be I get the error message that the TLS1.2 connection to the server could not be established. 

After countless hours of trial and error and issue reading on github it's safe to say that I am completely lost on this. Do you guys have any ideas what I might be doing wrong here? 

Many thanks in advance!

**Comment by andryyy** [comment]
Can you screenshot the settings you put in your phone manually? Feel free to censor the domain part. :)

Can you login to SOGo via web interface with these credentials?

Mit besten Grüßen
André Peters

**Comment by epon-was-taken** [comment]
![27a6fe25-d9d5-4842-ae50-5e502cfba19d](https://user-images.githubusercontent.com/50664702/61328410-42cfaa80-a81b-11e9-9027-a616cf8feead.jpg)

Hello andryyy,

thanks for your reply. Above you'll find the screenshot of the mail settings as autodiscover puts them (with the exeption that I entered my actual mail server and not just the domain name).

I can logon via SOGo with no problems. Also I can use my mail accounts on Apple Mail on OS X via the mobileconfig method without problems. 

Best Regards
Patrick

**Comment by epon-was-taken** [comment]
It's me again. I did some more troubleshooting and found out that I am served a completely wrong certificate on one of my domains (one of my vhosts that are also running on this server). 

So I looked again at the reverse proxy documentation and my nginx setup (and please correct me if I'm wrong). And if I understand it correctly, the reverse proxy hands off certificate requests to the mailcow acme container. 

So I looked up the logs with `docker-compose logs acme-mailcow`
What striked me first is this:

`Found AAAA record for autodiscover.domain1.de: IPv6Address - skipping A record check
Confirmed AAAA record with IP IPv6Address, but HTTP validation failed
Found AAAA record for autoconfig.domain1.de: IPv6Address - skipping A record check
Confirmed AAAA record with IP IPv6Address, but HTTP validation failed
Found AAAA record for autodiscover.domain2.de: IPv6Address - skipping A record check
Confirmed AAAA record with IP IPv6Address, but HTTP validation failed
Found AAAA record for autoconfig.domain2.de: IPv6Address - skipping A record check
Confirmed AAAA record with IP IPv6Address, but HTTP validation failed
Found AAAA record for mail.domain2.de: IPv6Address - skipping A record check
Confirmed AAAA record IPv6Address`

mail.domain2.de is my mailserver and this is the only DNS entry that has an AAAA Record set by me. autodiscover, autoconfig is just a CNAME entry. Do I actually need A Records for them? 

Second of all:
I've read somewhere on here, that I don't have to set additional SANs when I add a domain to my mailserver. The mailcow.conf comments state that I have to set them, when I add additional domains. 
What information is the right one? 


Many thanks in advance guys.

**Comment by mkuron** [comment]
> is just a CNAME entry. Do I actually need A Records for them?

It follows the CNAME, so that's fine.

> that I don't have to set additional SANs when I add a domain to my mailserver

You only need the autodiscover (for EAS) and autoconfig (for Thunderbird). They are added to the certificate automatically once you create the DNS records.

**Comment by epon-was-taken** [comment]
Well it's me again. And this time I actually solved the darn problem. And I feel positively stupid about it as it was a very simple mistake.

Because of what mkuron said I figured that the mailcow settings are all right. So I had a look at my nginx reverse proxy settings.

 location /Microsoft-Server-ActiveSync {
    proxy_pass http://127.0.0.1:8080/**Microsoft-Server-ActiveSync**;

The bold Microsoft-Server-ActiveSync was missing. It all works fine now. Thank you all for your support!
